cart.mjs

```diff
@@ -1,14 +1,117 @@
+    
+    let user = core.getUser(cartRow.user);
+
+
+
+    console.log(user,'\n');
+    console.log(prodottiUtente,'\n');*/
+
+    let user=core.getUser(cartRow.user);
+    let uuid=user.uuid;
+    let promo=user.promo;
+    let sconto=core.getPercentageFromPromoCode(promo)
+    let prodottiUtente=cartRow.products;
+    let nome=user.firstName;
+    let cognome=user.lastName;
+    let scontoEffettivo;
+    let totaleScontato;
+    let rimanenza;
+    let ricevuta='';
+
+
+
+   if (prodottiUtente.length>0){
+        let portafoglio=user.wallet;
+        let totale=0;
+        for (let oggetta of prodottiUtente){
+            let oggetto=core.getProduct(oggetta);
+            let prezzo=oggetto.price;
+            totale=prezzo+totale;
+        }
+        scontoEffettivo=sconto*totale;
+        totaleScontato=totale-scontoEffettivo;
+        rimanenza=portafoglio-totaleScontato;
+
+
+
+
+        if((portafoglio+scontoEffettivo)>totale){
+            //stampa inizio
+
+            var today  = new Date();
+            var options = { weekday: 'short', year: 'numeric', month: 'long', day: 'numeric' };
+            const data=`${today.toLocaleDateString("en-US", options)}`;
+
+            const dataOra=`${today.toLocaleTimeString("en-US",options)}`;
+            const trattino='-';
+            ricevuta+=`+${trattino.repeat(50)}+\n`
+            ricevuta+=`NOMEMACCHINA Cart - 43874\n`;
+            ricevuta+=`${data}\n`;
+            ricevuta+=`*${trattino.repeat(50)}*\n`;
+
+            //stampa codice, nome prodotto e prezzo
+            for(let a of prodottiUtente){
+                let oggetto=core.getProduct(a);
+                let prezzo=oggetto.price;
+                let codice=oggetto.ean;
+                let nomeoggetto=oggetto.name;
+                ricevuta+=core.maiuscoloParole(`[${codice}]\t${nomeoggetto}\t${prezzo.toFixed(2)}`);
+                ricevuta+=`\n`
+            }
+
+            //stampa totale
+            ricevuta+=`*${trattino.repeat(50)}*\n`;
+            ricevuta+=`totale: ${totale.toFixed(2)}\n`;
+            ricevuta+=`+${trattino.repeat(50)}+\n`;
+
+
+            //stampa sconto e credito
+            if (sconto!==0){
+                ricevuta+=`Sconto: ${scontoEffettivo.toFixed(2)}\n`;
+                ricevuta+=`Totale scontato: ${totaleScontato.toFixed(2)}\n`;
+                ricevuta+=` \n`;
+                ricevuta+=`CODICE PROMO: ${promo}\n`;
+                ricevuta+=`**${trattino.repeat(50)}**\n`;
+                ricevuta+=`${nome} ${cognome} ha un credito residuo ${rimanenza.toFixed(2)}\n`;
+        
+            }
+            else{
+                let rimanenza=portafoglio-totale;
+                ricevuta+=`${nome} ${cognome} ha un credito residuo ${rimanenza.toFixed(2)}\n`;
+            }
+
+
+
+            ricevuta+=``;
+            ricevuta+=`**${trattino.repeat(50)}**\n`;
+
+           
+            core.fs.writeFileSync(`./receipts/${uuid}_receipt_${data}.txt`,ricevuta);
```

Comment:
Non so dove tu l'abbia copiata ma funziona. `core.fs.writeFileSync()` mettere core davanti a fs però non è corretto. 
fs è di node nativa!! https://nodejs.org/api/fs.html#fswritefilesyncfile-data-options

Reply:
Cercando su internet ho trovato quella funzione per scrivere su un file.
In un progetto di prova ho scritto il seguente codice:
`var fs=require('fs');

const parola=`ciao`;
fs.writeFileSync(`saluto.txt`,parola)`
per vedere se funzionava e ha funzionato.

Nell'esercizio del carrello se scrivo 
`var fs=require('fs');`
mi da questo errore:

> var fs=require('fs');
       ^

ReferenceError: require is not defined in ES module scope, you can use import instead


Il motivo per cui avevo messo il core davanti alla funzione in cart.mjs è perche' esportavo dal core-cart `import * as fs from "fs";`

Reply:
stiamo usanto mjs invece di js gli import delle librerie di node vanno fatte cosi:
`import * as fs from "fs";` e non così `fs = require('fs');` in classe spiego a tutti il perchè di uno e dell'altro.

In ogni caso bravo.

Reply:
Per favore carica lo zip sulla piattaforma dell'its cosi ti do la valutazione!